### PR TITLE
stats: filter out robot events and add migrated fields

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

File: cds_migrator_kit/rdm/migration/stats/event_generator.py [hunk 1]
```diff
@@ -11,6 +11,25 @@
...
+    # filter out robot events
+    is_bot = entry.get("bot")
+    before_COUNTER = False
+
+    if is_bot is None:
+        # We migrate them as valid and we mark them for future discoverability
+        before_COUNTER = True
+        is_bot = False
+
+    return is_bot, before_COUNTER
```
Comment (ntarocco) [inline cds_migrator_kit/rdm/migration/stats/event_generator.py:30]: Super minor: if you want to be even more explicit

```suggestion
    is_bot_missing = "bot" in entry
    if is_bot_missing:
        # when is_bot is missing, we assume that it was a human
        # but we mark this uncertainty with before_COUNTER =True
        is_bot = False
        before_COUNTER = True
    else:
       is_bot = entry["bot"]
       before_COUNTER = True

    return is_bot, before_COUNTER
```

File: cds_migrator_kit/rdm/migration/stats/event_generator.py [hunk 2]
```diff
@@ -96,13 +118,17 @@ def process_download_event(entry, rec_context, logger):
...
-        "is_robot": entry.get("bot", False),
         "country": entry.get("country", ""),
         "visitor_id": entry.get("visitor_id", ""),
         "unique_session_id": entry["unique_session_id"],
         # Note: id_bibrec doesn't have the new format pids
         "unique_id": f"ui_{_record_version['new_recid']}",
+        # Mark the event as migrated
+        "is_lcds": True,
+        # Mark if event was COUNTER compliant
+        "before_COUNTER": before_COUNTER,
```
Comment (ntarocco) [inline cds_migrator_kit/rdm/migration/stats/event_generator.py:131]: did we say that we add it only when `before_COUNTER = true`?
New events will not have this field at all.

Comment (zzacharo) [reply cds_migrator_kit/rdm/migration/stats/event_generator.py:131]: Discussed IRL to keep it for differentiating between COUNTER compliant migrated events

File: cds_migrator_kit/rdm/migration/cli.py [hunk 3]
```diff
@@ -67,8 +67,12 @@ def stats():
     "--filepath",
     help="Path to the list of records file that the legacy statistics will be migrated.",
 )
+@click.option(
+    "--less-than-date",
+    help="ISO string e.g 2024-12-13T20:00:00 to migrate events up to this date.",
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/cli.py:72]: just for my knowledge update, which date will we use finally? the time we applied the redirection or the migration time?

File: cds_migrator_kit/rdm/migration/stats/event_generator.py [hunk 4]
```diff
@@ -11,6 +11,25 @@
...
+def flag_robots_and_COUNTER(entry):
+    """Return a tuple of booleans.
+
+    First element is the value of is_bot and the second the flag to mark the event as
+    `before_COUNTER` as we don't have information if the registered event was from a bot
+    or not.
+    """
+    is_bot_missing = "bot" in entry
+    if is_bot_missing:
+        # when is_bot is missing, we assume that it was a human
```
Comment (kpsherva) [inline cds_migrator_kit/rdm/migration/stats/event_generator.py:23]: nice comments verbosity :)

**Review by kpsherva — APPROVED** [review]

File: cds_migrator_kit/rdm/migration/cli.py (hunk 3 above)
Comment (zzacharo) [reply cds_migrator_kit/rdm/migration/cli.py:74]: TBD but I think it should be the redirection time

**Pull request merged**